### ISSUE TITLE
Metrics to assess core fragmentation

### ISSUE THREAD

**Issue by metazool** [BritishGeologicalSurvey/CoreScore#4]
From Zayad and Mark's original paper/outline we have a set of core quality metrics that assess fragmentation (could be expanded).

* https://scikit-image.org/docs/dev/api/skimage.measure.html#skimage.measure.regionprops to produce measures for core fragments as we did for labelled pores in thin sections
* API endpoint to return metrics based on estimated labels

**Comment by metazool** [comment]
Existing rock quality designation scheme / "gold standard" engineering geologist assessment of sample images

**Comment by mobiuscreek** [comment]
Sorry made it a duplicate here #9 but I'll close it and add my comment for reference:

> We currently have a single metric to assess rock quality that is implemented in fastaiv1 as foreground_acc and assesses the accuracy of the foreground (rock) segmentation. We should revisit this and implement more metrics. This can be done by having multiple metrics, e.g starting from the crude, like the total foreground accuracy and then measure the accuracy for each respective class. We should also add existing core quality assessments e.g the Rock Quality Designation (RQD) with the caveat that the segmentation has to be thoroughly tested for its performance before we have concrete results.

**Comment by mftf** [comment]
Yes the end goal would be to reassess the metrics we use to designate core quality against something like RQD. However RQD is recorded usually as core is collected and determines whether features are natural or induced either by drilling or handling. It is also collected over a 3m interval. 

This is different to core score which, in the first instance aims to provide an overview of the core condition today which can be years to decades after it was collected so the two are not directly comparable. However we should look for examples where we can compare the two. 

Core score is not about replacing RQD just acknowledging that core degrades over time and allowing users to assess core quality prior to undertaking analysis.

**Comment by metazool** [comment]
Having read back the original study we are also looking at a number of measurements that go into deriving the core quality index from the labelled areas:

* Number of fragments
* Mean fragment area
* Minimum and maximum fragment area
* Total fragment perimeter

So we may be talking about different things?  And #9 is then still a valid issue for metrics related to model performance at core identification rather than metrics related to core usability?

**Comment by mftf** [comment]
Yes I would say the issue #9 around model metrics is a valid one as it differs from the core quality index, though I am not sure I have much to contribute on that. The parameters you mention are the ones that Zayad and I came up with as things which would be useful for assessing core quality. 

One thing that the tool does not allow is identification of planer features such as core breaks. However we can estimate this from the number of fragments e.g. if you have three fragments then you would expect two core breaks. 

If there are additional parameter that you think might be useful then let me know
